File: Data Structures & Algorithms/find-pivot-index/submission-1.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        '''
        Time Complexity: O(n) n = len(nums)
        Space Complexity: O(1)
        '''
        total = sum(nums)
        lsum = 0
        for i in range(len(nums)):
            rsum = total - nums[i] - lsum
            if lsum == rsum:
                return i
            lsum += nums[i]
        return -1

File: Data Structures & Algorithms/sort-colors/submission-1.cpp
#include <array>

class Solution {
public:
    void sortColors(vector<int>& nums) {
        std::array<int, 3> count = {};
        for (int num : nums) {
            count[num] ++;
        }

        int i = 0;
        for (int c = 0; c < count.size(); c++) {
            for (int j = 0; j < count[c]; j++) {
                nums[i] = c;
                i++;
            }
        }
    }
};